From f47e8c6f9e461803468a64581d3640d873352eaa Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Thu, 1 Feb 2018 10:14:30 +0800
Subject: [PATCH] qt/python: Add variables to tests

* configure.ac, lang/python/Makefile.am:
  New variable to `lang/python', set to `lang/python' if RUN_LANG_PYTHON_TESTS

* configure.ac, lang/qt/Makefile.am:
  New variable to `lang/qt', set to `lang/qt' if RUN_LANG_QT_TESTS

Upstream-Status: Submitted [gnupg-devel@gnupg.org]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 configure.ac            | 11 +++++++++++
 lang/python/Makefile.am |  3 +++
 lang/qt/Makefile.am     |  8 +++++++-
 3 files changed, 21 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 608c376..fad7467 100644
--- a/configure.ac
+++ b/configure.ac
@@ -536,6 +536,17 @@ AC_ARG_ENABLE(g13-test,
          run_g13_test=$enableval)
 AM_CONDITIONAL(RUN_G13_TESTS, test "$run_g13_test" = "yes")
 
+run_lang_python_test="yes"
+AC_ARG_ENABLE(lang-python-test,
+  AC_HELP_STRING([--disable-lang-python-test], [disable Python regression test]),
+         run_lang_python_test=$enableval)
+AM_CONDITIONAL(RUN_LANG_PYTHON_TESTS, test "$run_lang_python_test" = "yes")
+
+run_lang_qt_test="yes"
+AC_ARG_ENABLE(lang-qt-test,
+  AC_HELP_STRING([--disable-lang-qt-test], [disable Qt regression test]),
+         run_lang_qt_test=$enableval)
+AM_CONDITIONAL(RUN_LANG_QT_TESTS, test "$run_lang_qt_test" = "yes")
 
 # Checks for header files.
 AC_CHECK_HEADERS_ONCE([locale.h sys/select.h sys/uio.h argp.h
diff --git a/lang/python/Makefile.am b/lang/python/Makefile.am
index 8d74cbd..43b5d1e 100644
--- a/lang/python/Makefile.am
+++ b/lang/python/Makefile.am
@@ -24,7 +24,10 @@ EXTRA_DIST = \
 	examples \
 	src
 
+if RUN_LANG_PYTHON_TESTS
 SUBDIRS = . tests
+endif
+
 
 .PHONY: prepare
 prepare: copystamp
diff --git a/lang/qt/Makefile.am b/lang/qt/Makefile.am
index ab85960..a244ede 100644
--- a/lang/qt/Makefile.am
+++ b/lang/qt/Makefile.am
@@ -19,6 +19,12 @@
 # Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
 # 02111-1307, USA
 
-SUBDIRS = src tests doc
+if RUN_LANG_QT_TESTS
+qttests = tests
+else
+qttests =
+endif
+
+SUBDIRS = src ${qttests} doc
 
 EXTRA_DIST = README
-- 
1.8.3.1

